custom-tests.json

```diff
@@ -7,6 +7,24 @@
       "drawElementsInstancedANGLE": "return a && 'drawElementsInstancedANGLE' in a;",
       "vertexAttribDivisorANGLE": "return a && 'vertexAttribDivisorANGLE' in a;",
       "VERTEX_ATTRIB_ARRAY_DIVISOR_ANGLE": "return a && 'VERTEX_ATTRIB_ARRAY_DIVISOR_ANGLE' in a;"
+    },
+    "AudioBuffer": {
+      "__base": "var audioCtx = new (window.AudioContext || window.webkitAudioContext)(); if (!audioCtx) {return false}; var buffer = audioCtx.createBuffer(2, audioCtx.sampleRate * 3, audioCtx.sampleRate);",
+      "__test": "return !!buffer;",
+      "duration": "return buffer && 'duration' in buffer;",
```

Comment:
These look very regular. It looks like the important part of this is how to create an instance of `AudioBuffer` and the rest could be generated. I think this will be a very common pattern, and that we would benefit for having a notion of factories, i.e., code that creates instances.

Some factories can be generated from interpreting Web IDL, like that `navigator.mediaDevices` is a `MediaDevices` instance. But then some should be custom factories.

No need to fix this now, just dropping some ideas.

Reply:
Yeah, that makes sense!  I'll follow up with another PR to improve the custom test generation.